File: Lecture08/MoreFragments/app/src/main/java/com/codingblocks/morefragments/MainActivity.java
package com.codingblocks.morefragments;

import android.content.Intent;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    ViewPager viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        viewPager = (ViewPager) findViewById(R.id.viewPager);


        ((Button) findViewById(R.id.btnTabAct))
                .setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        startActivity(
                                new Intent(MainActivity.this, TabbedActivity.class)
                        );
                    }
                });

        viewPager.setAdapter(new MyFragmentPagerAdapter(
                getSupportFragmentManager(),
                10
        ));
        viewPager.setOffscreenPageLimit(3);
    }
}

File: Lecture14/Database/app/src/main/java/com/codingblocks/database/MainActivity.kt
package com.codingblocks.database

import android.database.sqlite.SQLiteDatabase
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.ArrayAdapter
import com.codingblocks.database.db.TodoDatabaseHelper
import com.codingblocks.database.db.tables.TodoTable

import kotlinx.android.synthetic.main.activity_main.*;

class MainActivity : AppCompatActivity() {
    var db: SQLiteDatabase? = null


    //var todos: ArrayList<Todo> = ArrayList<Todo>()
    var todos: ArrayList<String> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        db = TodoDatabaseHelper(this).writableDatabase
        todos = TodoTable.getTodos(db)


        var todoAdapter = ArrayAdapter<String>(
                this,
                android.R.layout.simple_expandable_list_item_1,
                android.R.id.text1,
                todos

        )
        lvTodos.adapter = todoAdapter


        btnAddTodo.setOnClickListener {
            TodoTable.addTodo(db, etNewTodo.text.toString())

            //todos.add(etNewTodo.text.toString()) //Only add the new task to view layer
            todos.clear();
            todos.addAll(TodoTable.getTodos(db))  // Refresh the entire list
            Log.d(TAG, "" + todos.size)

            todoAdapter.notifyDataSetChanged()
        }

    }

    companion object {
        val TAG = "TODO"
    }
}
